src/scss/_grid.scss

```diff
@@ -8,8 +8,98 @@
 	}
 
 	.o-layout {
-		@include _oLayoutGridBase;
 		@include oTypographySize(1);
+		position: relative;
+		display: grid;
+		box-sizing: border-box;
 		min-height: 100vh;
+		min-width: 100%;
+		grid-template-columns: 100%;
+		grid-template-rows: auto 1fr auto;
+		grid-gap: $_o-layout-gutter;
+		grid-template-areas:
+			"header"
+			"main"
+			"footer";
+	}
+}
+
+
+/// Grid for a landing page layout (e.g. homepage).
+/// @access private
+@mixin _oLayoutLandingGrid() {
+	.o-layout.o-layout--landing {
+		@include oGridRespondTo($from: M) {
+			grid-template-areas:
+				"header header header"
+				". main ."
+				"footer footer footer";
+			grid-template-columns: 1fr minmax(auto, $_o-layout-container-max-width) 1fr;
+		};
+	}
+}
+
+/// Grid for document layout.
+/// @access private
+@mixin _oLayoutDocsGrid() {
+	.o-layout.o-layout--docs {
+		grid-template-rows: auto auto 1fr auto;
+		grid-template-areas:
+			"header"
+			"sidebar"
+			"main"
+			"footer";
+
+		//sass-lint:disable mixins-before-declarations - Source order matters for media queries. Although @at-root is used anyway, which will always output the media after.
+		@include oGridRespondTo($from: M) {
+			grid-template-rows: auto 1fr auto;
+			grid-template-columns: 1fr minmax(auto, $_o-layout-aside-max-width) $_o-layout-main-section-width minmax(20ch, $_o-layout-aside-max-width) 1fr;
+			grid-template-areas:
+				"header header header header header"
+				". sidebar main main ."
+				"footer footer footer footer footer";
+		};
+		//sass-lint:enable mixins-before-declarations
+	}
+
+	.o-layout.o-layout--docs > .o-layout__main {
+		@include oGridRespondTo($from: M) {
+			display: grid;
+			grid-column-gap: $_o-layout-gutter;
+			grid-template-columns: $_o-layout-main-section-width 1fr;
+			align-content: flex-start;
+			padding: 0;
+		}
+
+		> * {
+			grid-column: 1;
+		}
+
+		@include oGridRespondTo($from: L) {
+			> aside {
+				grid-column: 2;
+				grid-row: span 3; //hack mchack, need to look into spanning across rows irrespective of siblings
```

Comment:
😂 i'm amused this has remained

Reply:
😂 I'll see if there's anything we can do to remove hack mchack.